CRM/Activity/BAO/Activity.php

```diff
@@ -1638,73 +1638,81 @@ public static function sendEmail(
   }
 
   /**
-   * Send SMS.
+   * Send SMS.  Returns: bool $sent, int $activityId, int $success (number of sent SMS)
    *
    * @param array $contactDetails
    * @param array $activityParams
-   * @param array $smsParams
-   * @param $contactIds
-   * @param int $userID
+   * @param array $smsProviderParams
+   * @param array $contactIds
+   * @param int $sourceContactId This is the source contact Id
    *
-   * @return array
+   * @return array(bool $sent, int $activityId, int $success)
    * @throws CRM_Core_Exception
    */
   public static function sendSMS(
-    &$contactDetails,
+    &$contactDetails = NULL,
     &$activityParams,
-    &$smsParams = array(),
-    &$contactIds,
-    $userID = NULL
+    &$smsProviderParams = array(),
+    &$contactIds = NULL,
+    $sourceContactId = NULL
   ) {
-    if ($userID == NULL) {
-      $userID = CRM_Core_Session::getLoggedInContactID();
+    if (!isset($contactDetails) && !isset($contactIds)) {
+      Throw new CRM_Core_Exception('You must specify either $contactDetails or $contactIds');
+    }
+    // Populate $contactDetails and $contactIds if only one is set
+    if (is_array($contactIds) && !empty($contactIds) && empty($contactDetails)) {
```

Comment:
if `$contactIds` is not an array what will it be just a single id or will it be a `,` separated string of ids? could we cast to an array here so that we use the foreach anyway and same comment but about `$contactDetails` in L1673

Reply:
Historically both $contactIds and $contactDetails had to be set which was a bit silly. This PR does away with that.  We are assuming that both have to be passed in as an array (and that's specified in the docblock) - I think that is sensible, rather than trying to accept non-array values.  Ideally we could drop one ($contactDetails) altogether but that means changing the function signature which will break stuff. @michaelmcandrew checking you agree.

Reply:
hey there, no strong feelings. i guess seamus's comment stems from the fact that you are using is_array(). I am presuming you meant it in the sense of  'is array as opposed to null'. In other words, seems fine as is to me, but I coudl be missing some context.